python/cudf/cudf/tests/test_joining.py

```diff
@@ -69,6 +69,35 @@ def pd_odd_joins(left, right, join_type):
         return left[left.index.isin(right.index)][left.columns]
 
 
+def assert_join_results_equal(expect, got, how, **kwargs):
+    if how == "right":
```

Comment:
First of all, 🔥 . 

I noticed though that this only seems to do anything differently for this one kwarg, but below when this is being used `outer` and `left` are passed often. Even if there is no intended behavior difference for the other kinds of merges, I'd still make it so that the keyword has effect otherwise I might be able to pass `how="blah"`.

Reply:
Added validation for `how`.